lib/autoupdate/utils.go

```diff
@@ -0,0 +1,168 @@
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package autoupdate
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"errors"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-semver/semver"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
+	"github.com/gravitational/teleport/lib/utils"
+)
+
+const (
+	// FlagEnt represents enterprise version.
+	FlagEnt = 1 << 0
+	// FlagFips represents enterprise version with fips feature enabled.
+	FlagFips = 1 << 1
+)
+
+var (
+	// featureFlag stores information about enable
+	featureFlag int
+)
+
+// ToolsDir returns the path to {tsh, tctl} in $TELEPORT_HOME/bin.
+func ToolsDir() (string, error) {
+	home := os.Getenv(types.HomeEnvVar)
+	if home == "" {
+		var err error
+		home, err = os.UserHomeDir()
+		if err != nil {
+			return "", trace.Wrap(err)
+		}
+	}
+
+	return filepath.Join(filepath.Clean(home), ".tsh", "bin"), nil
+}
+
+func version(toolsDir string) (string, error) {
```

Comment:
Recommend longer global function name that won't be shadowed by variables called `version`.